examples/s2_field_combination_extern_rf_train/nrw_crop_extern_s2_workflow_to_datasets_with_timeshift.ipynb

```diff
@@ -0,0 +1,2364 @@
+   "source": [
+    "used_quota_start = ge.get_quota(user_id)['used']\n",
+    "used_quota_start"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set the area of interest. It is defined as a bounding box in UTM 32 N (EPSG:32632).\n",
+    "It is locted in NRW, Germany and covers the area between Willingen, Lippstadt and Werl."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(421395, 5681078, 476201, 5727833)"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "bounds_array = [421395,  5681078, 476201, 5727833]\n",
+    "xmin = bounds_array[0]\n",
+    "ymin = bounds_array[1]\n",
+    "xmax = bounds_array[2]\n",
+    "ymax = bounds_array[3]\n",
+    "\n",
+    "(xmin, ymin, xmax, ymax)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Using the bounding box, a time interval and a resolution, we define the area of interest as a temporal raster space-time cube."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'spatialBounds': {'upperLeftCoordinate': {'x': 421395, 'y': 5727833},\n",
+       "  'lowerRightCoordinate': {'x': 476201, 'y': 5681078}},\n",
+       " 'timeInterval': {'start': '2021-01-01T00:00:00.000+00:00',\n",
+       "  'end': '2021-12-31T00:00:00.000+00:00'},\n",
+       " 'spatialResolution': {'x': 10.0, 'y': 10.0}}"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from datetime import datetime\n",
+    "time_start = datetime(2021, 1, 1)\n",
+    "time_end = datetime(2021, 12, 31)\n",
+    "\n",
+    "study_area = ge.api.RasterQueryRectangle(\n",
+    "    spatialBounds=ge.SpatialPartition2D(xmin, ymin, xmax, ymax).to_api_dict(),\n",
+    "    timeInterval=ge.TimeInterval(time_start, time_end).to_api_dict(),\n",
+    "    spatialResolution=ge.SpatialResolution(10.0, 10.0).to_api_dict(),\n",
+    ")\n",
+    "study_area"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For each band (B02, B03, B04, B08) as well as the scene mask (SCL), we create a workflow, that downloads the Sentinel-2 data for the area of interest and stores it as a new dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{}"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "status_download = {}\n",
+    "\n",
+    "for b in [\"B02\", \"B03\", \"B04\", \"B08\", \"SCL\"]:\n",
```

Comment:
Wouldn't this be faster as a fork-join of everything rather than sequentially downloading it?

Reply:
i guess this depends on the bottleneck. If it is the Download then downloading more things in parallel will not really help.